src/dlsnode/util/aio/AsyncIO.d

```diff
@@ -204,6 +205,74 @@ class AsyncIO
         }
     }
 
+    /***************************************************************************
+
+        Set of non-blocking methods. The user is responsible to suspend the fiber,
+        and AsyncIO will not resume it. Instead, the callback will be called
+        where the user can do whatever is required.
+
+        TODO: this is a class, since there's no outer.
+
+    ***************************************************************************/
+
+    public class Nonblocking
+    {
+        /**************************************************************************
+
+            Issues a pread request, filling the buffer as much as possible,
+            expecting the user to suspend the caller manually.
+
+            This will read buf.length number of bytes from fd to buf, starting
+            from offset.
+
+            Params:
+                buf = buffer to fill
+                ret_val = return value to fill
+                fd = file descriptor to read from
+                offset = offset in the file to read from
+                finish_callback_dg = delegate to call when the request has finished,
```

Comment:
Naming nitpick: there's probably no need to point out that it's a callback _and_ a delegate.